akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphPartitionSpec.scala

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream.testkit._
+import akka.stream.{ OverflowStrategy, ActorMaterializer, ActorMaterializerSettings, ClosedShape }
+import akka.stream.testkit.Utils._
+import scala.concurrent.duration._
+
+class GraphPartitionSpec extends AkkaSpec {
+
+  val settings = ActorMaterializerSettings(system)
+    .withInputBuffer(initialSize = 2, maxSize = 16)
+
+  implicit val materializer = ActorMaterializer(settings)
+
+  "A partition" must {
```

Comment:
In general there should be two kinds of tests:

* functional tests that do not use any testkit functionality and just verify that the right elements eventually end up in the right places
* specific tests around completion and cancellation that can only be done using `TestSink.probe` and `TestSource.probe`